### PR TITLE
Collapse all RunnerMocks into only one: RunnerOrderedMock

### PR DESCRIPTION
**Pull request by jgsogo**
Remove several RunnerMock classes that were being declared in the code and use only one: `RunnerOrderedMock`. This class allows declaring a command sequence and the return code of each one of them, and then it checks that commands are being called in the given order.

Only test code is affected

### CODE REVIEW

File: conans/test/build_helpers/autotools_configure_test.py [hunk 1]
```diff
@@ -473,94 +473,89 @@ def cross_build_command_test(self):
                                                 "compiler.libcxx": "libstdc++"}),
                                   None, runner)
         ab = AutoToolsBuildEnvironment(conanfile)
-        ab.configure()
         if platform.system() == "Windows":
-            # Not crossbuilding
-            self.assertFalse(ab.host)
-            self.assertFalse(ab.build)
```
Comment (lasote) [inline conans/test/build_helpers/autotools_configure_test.py]: Are not these checks done anymore?

Comment (jgsogo) [reply conans/test/build_helpers/autotools_configure_test.py]: No, the command in Windows has no `--host` neither `--build` arguments... I think there is no need to test it here.

**Comment by jgsogo** [comment]
Before reviewing, we must agree if this workflow is what we want:
1. Create `runner`
2. Add commands expected (with return types)
3. Run code

There is something I don't like of this approach: there is no `assert/test` sorrounding the code, so it seems you are not testing anything :/

**Comment by lasote** [comment]
I don't know, I think it could limit the test capacity because of not being able to assert in the middle.
I would keep the current mess.

**Comment by jgsogo** [comment]
You can assert in the middle: add some commands, run test to one point, add more commands, continue,...

**Pull request closed without merging**